src/s3_fetchers.rs

```diff
@@ -1,6 +1,49 @@
+use async_trait::async_trait;
 use std::str::FromStr;
 
-use aws_sdk_s3::Client;
+use aws_sdk_s3::{
+    output::{get_object_output, GetObjectOutput},
+    Client,
+};
+
+#[async_trait]
+pub trait GetObject {
+    async fn get_object(
+        &self,
+        bucket: &str,
+        prefix: &str,
+    ) -> Result<GetObjectOutput, aws_sdk_s3::types::SdkError<aws_sdk_s3::error::GetObjectError>>;
+}
+
+#[derive(Clone, Debug)]
+pub struct S3GetObject {
```

Comment:
I definitely don't like the name, though I believe it's a work-in-progress naming for now. In case it isn't, I would suggest `LakeS3Client` for the wrapper name. Also, I would expect it to impl all the necessary stuff: `GetObject` and `ListObjects` to avoid having a bunch of wrappers.

P.S. I truly believe you keep that in mind already. Leaving such a comment just in case.

Reply:
I was mostly just copying the AWS tutorial 😄. I've gone with what you've suggested 👍🏽